### PR TITLE
SPNPTL-100 Fix Change Pending appears forever bug

### PR DESCRIPTION
**Pull request by vertein**
Before:

image

image

After:

image

image

### CODE REVIEW

**Comment by timlevett** [comment]
:+1:

File: student-preferred-name-portlet-webapp/src/main/java/edu/wisc/portlet/preferred/service/PreferredNameServiceImpl.java [hunk 1]
```diff
@@ -59,16 +59,20 @@ public String getStatus(PreferredName ldapPn) {
...
-            ) {
+            )){
             return "(not set)";
-        } else {
-
-            if (ldapPn.equals(jdbcPn)) {
+        }else if(jdbcPn == null || 
+                (StringUtils.isEmpty(jdbcPn.getFirstName()) && 
+                 StringUtils.isEmpty(jdbcPn.getMiddleName()) && 
+                 StringUtils.isEmpty(jdbcPn.getLastName()))){
```
Comment (apetro) [inline student-preferred-name-portlet-webapp/src/main/java/edu/wisc/portlet/preferred/service/PreferredNameServiceImpl.java]: Looks like a `PreferredName` can represent a preferred name deletion, and than it does so by the first, middle, and last name fields being empty.  Could refactor this logic into a `isRepresentingDeletion()` property on the `PreferredName` object, which would be eminently unit testable and might feel more object oriented.

File: student-preferred-name-portlet-webapp/src/main/java/edu/wisc/portlet/preferred/service/PreferredNameServiceImpl.java [hunk 2]
```diff
@@ -59,16 +59,20 @@ public String getStatus(PreferredName ldapPn) {
...
-            if (ldapPn.equals(jdbcPn)) {
+        }else if(jdbcPn == null || 
+                (StringUtils.isEmpty(jdbcPn.getFirstName()) && 
+                 StringUtils.isEmpty(jdbcPn.getMiddleName()) && 
+                 StringUtils.isEmpty(jdbcPn.getLastName()))){ 
+            return "(deletion pending)";
+        }else{
+            String ldapPreferredName = new StringBuilder().append(ldapPn.getFirstName()).append(ldapPn.getMiddleName()).toString();
+            String jdbcPreferredName = new StringBuilder().append(jdbcPn.getFirstName()).append(jdbcPn.getMiddleName()).toString();
+            if(ldapPreferredName.equals(jdbcPreferredName)){
```
Comment (apetro) [inline student-preferred-name-portlet-webapp/src/main/java/edu/wisc/portlet/preferred/service/PreferredNameServiceImpl.java]: This logic is not detecting when a change to only the last name is pending -- that edge case will be experienced as no change pending.  Backlogged as [SPNPTL-102](https://jira.doit.wisc.edu/jira/browse/SPNPTL-102).

May be worth an inline code comment to this effect to guide the reader in what this logic is doing and in particular why it is not considering last name.

**Comment by vertein** [comment]
Added an inline comment per suggestion and see new issue for improved functionality.

Comment about isEmpty preferred name acknowledged, but not going to dive into that in this bug fix.  :)

**Pull request merged**